yesod-auth/Yesod/Auth/Email.hs

```diff
@@ -299,18 +301,37 @@ getRegisterR = registerHandler
 -- Since: 1.2.6
 defaultRegisterHandler :: YesodAuthEmail master => AuthHandler master Html
 defaultRegisterHandler = do
-    email <- newIdent
-    tp <- getRouteToParent
+    ((f,widget),e) <- lift $ runFormPost registrationForm
+    toParentRoute <- getRouteToParent
     lift $ authLayout $ do
         setTitleI Msg.RegisterLong
         [whamlet|
             <p>_{Msg.EnterEmail}
-            <form method="post" action="@{tp registerR}">
+            <form method="post" action="@{toParentRoute registerR}">
                 <div id="registerForm">
-                    <label for=#{email}>_{Msg.Email}:
-                    <input ##{email} type="email" name="email" width="150" autofocus>
+                    ^{widget}
                 <button .btn>_{Msg.Register}
         |]
+    where
+        registrationForm extra = do
+            let emailSettings = FieldSettings {
+                fsLabel = "Email" ,
+                fsTooltip = Nothing ,
+                fsId = Just "email",
+                fsName = Just "email",
+                fsAttrs = []
+            }
+
+            (emailRes, emailView) <- mreq textField emailSettings Nothing
```

Comment:
This is using `textField`, which creates an [input of type `text`](http://hackage.haskell.org/package/yesod-form-1.4.6/docs/Yesod-Form-Fields.html#v:textField). The previous code created an input of type `email`, which has advantages like:

* Browsers can validate the email address before form submission
* CSS can select email form inputs
* Mobile browsers can display an email-specific keyboard

I think [`emailField`](http://hackage.haskell.org/package/yesod-form-1.4.6/docs/src/Yesod-Form-Fields.html#emailField) should be used so that the form is of the correct type.